improver_tests/spotdata/test_NeighbourSelection.py

```diff
@@ -668,16 +668,54 @@ def test_global_attribute(self):
 
     def test_wmo_ids(self):
         """Test that the returned cube has the wmo_ids present when they are
-        available. Should be None when they are not provided."""
+        available. Should be None when they are not provided. WMO IDs are
+        stored as 5 digits encoded as strings, so zero-padding is expected."""
 
         plugin = NeighbourSelection()
+        sites = self.global_sites + [
+            self.global_sites.copy()[0].copy(),
+            self.global_sites.copy()[0].copy(),
+        ]
+        sites[1].pop("wmo_id")
+        sites[2]["wmo_id"] = None
+        expected = ["00001", "None", "None"]
+        result = plugin.process(sites, self.global_orography, self.global_land_mask)
+
+        self.assertArrayEqual(result.coord("wmo_id").points, expected)
+
+    def test_use_of_unique_ids(self):
+        """Test that the returned cube has the unique_id present when they are
+        provided. These are stored as 8 digits encoded as strings, so
+        zero-padding is expected."""
+
+        plugin = NeighbourSelection(unique_site_id_name="met_office_site_id")
         sites = self.global_sites + [self.global_sites.copy()[0].copy()]
+        sites[0]["met_office_site_id"] = sites[0]["wmo_id"]
         sites[1]["wmo_id"] = None
-        expected = ["1", "None"]
-
+        sites[1]["met_office_site_id"] = 353
+        expected = ["00000001", "00000353"]
         result = plugin.process(sites, self.global_orography, self.global_land_mask)
 
-        self.assertArrayEqual(result.coord("wmo_id").points, expected)
+        self.assertArrayEqual(result.coord("met_office_site_id").points, expected)
+
+    def test_error_for_incomplete_unique_ids(self):
+        """Test that an error is raised if the list of unique IDs is incomplete,
+        or if it contains duplicate IDs."""
+
+        plugin = NeighbourSelection(unique_site_id_name="met_office_site_id")
+        sites = self.global_sites + [self.global_sites.copy()[0].copy()]
+        sites[0]["met_office_site_id"] = sites[0]["wmo_id"]
+
+        # unique_site_id not set for every site
+        msg = "The unique_site_id is not available for every site"
+        with self.assertRaisesRegex(ValueError, msg):
+            plugin.process(sites, self.global_orography, self.global_land_mask)
+
+        # duplicate site id used
+        sites[1]["met_office_site_id"] = sites[0]["wmo_id"]
+        msg = "The unique_site_id is not unique for every site"
```

Comment:
The first test trips the (now) KeyError because sites[1] has no met_office_site_id; sites[0] is fine. For the second test sites[1] has the met_office_site_id set, so the KeyError is not raised. However sites[0] and sites[1] share the same met_office_site_id, hence it trips the duplicate id check.